easybuild/easyblocks/generic/cmakemake.py

```diff
@@ -320,6 +336,22 @@ def configure_step(self, srcdir=None, builddir=None):
 
         return out
 
+    def set_cmake_env_vars_python(self):
```

Comment:
I'm a bit confused by this one, what's the plan with this?

As is, it can only be used in easyblocks that derive from `CMakeMake`?

Reply:
Yes I would make 2 free functions out of that:
1) Return a configure string with `-DPython*_EXECUTABLE=...`, maybe optionally only as a dict
2) Return these env variables

The first function is the safe way for every situation we can pass CMake variables. We can use that to enhance the GROMACS easyblock to move the CMake config options out from the EasyConfig to the easyblock and add those options.   

The 2nd function can be used whenever we cannot use the first. It *might* still pick up a `python` from a virtualenv when `Python*_FIND_STRATEGY` isn't set to `LOCATION` ("older" codes default this to LOCATION if unset) but at least we did a best effort to avoid that which might work for most cases.

Reply:
Added the 2 separate functions to return the options as a dict or as a string:

- `get_cmake_python_config_dict`
- `get_cmake_python_config_str`

 and changed the name of the function to set the env vars to `set_cmake_python_env_hints`  (https://github.com/easybuilders/easybuild-easyblocks/pull/3463/commits/9a31f79bf8c3c4d9bb5e18ecf6d6f6f21e57d486)

There might be an argument of making that function a context manager to be used with `with`